lib/queue.js

```diff
@@ -551,6 +551,13 @@ Queue.prototype.processStalledJob = function(job){
           if(!isMember){
             return _this.processJob(job);
           }
+
+          return job.releaseLock(_this.token);
+        }).catch(function(err) {
```

Comment:
It seems like this would be appropriate for a `finally` block.

However, Bluebird actually [recommends](http://bluebirdjs.com/docs/api/resource-management.html) using something called "disposers" to ensure that the lock would be released even if the `finally` block was not called due to an internal failure in the above promise chain.